tests/testGalSimInterface.py

```diff
@@ -264,6 +344,73 @@ def catalogTester(self, catName=None, catalog=None, nameRoot=None):
             self.assertTrue(unDrawnDetectors<2)
             self.assertTrue(drawnDetectors>0)
 
+
+    def compareCatalogs(self, cleanCatalog, noisyCatalog, gain, readnoise):
+        """
+        Read in two catalogs (one with noise, one without).  Compare the flux in each image
+        pixel by pixel.  Make sure that the variation between the two is within expected limits.
+
+        @param [in] cleanCatalog is the noiseless GalSimCatalog instantiation
+
+        @param [in] noisyCatalog is the noisy GalSimCatalog instantiation
+
+        @param [in] gain is the electrons per ADU for the GalSimCatalogs
+
+        @param [in] readnoise is the electrons per pixel per exposure of the GalSimCatalogs]
+        """
+
+        cleanFileList, cleanBandpassDict = self.getFilesAndBandpasses(cleanCatalog, nameRoot='clean')
+        noisyFileList, noisyBandpassDict = self.getFilesAndBandpasses(noisyCatalog, nameRoot='unclean')
+
+        #calculate the expected skyCounts in each filter
+        backgroundCounts = {}
+        for filterName in noisyBandpassDict.keys():
+            cts = expectedSkyCountsForM5(noisyCatalog.obs_metadata.m5[filterName], noisyBandpassDict[filterName],
+                                         noisyCatalog.photParams, seeing=noisyCatalog.obs_metadata.seeing[filterName])
+
+            backgroundCounts[filterName] = cts
+
+        # Go through each image pixel by pixel.
+        # Treat the value in the clean image as the mean intensity for that pixel.
+        # Sum up (noisy-clean)^2/var
+        # where var is determined by Poisson statistics from mean and readnoise.
+        # Divide by the number of pixel
+        # Make sure that this average does not deviate from unity
+
+        countedImages = 0
+        for noisyName, cleanName in zip(noisyFileList, cleanFileList):
+            noisyIm = afwImage.ImageF(noisyName).getArray()
+            cleanIm = afwImage.ImageF(cleanName).getArray()
+
+            totalVar = 0.0
+            totalMean = 0.0
+            ct = 0.0
+
+            self.assertEqual(cleanIm.shape[0], noisyIm.shape[0], msg='images not same shape')
+            self.assertEqual(cleanIm.shape[1], noisyIm.shape[1], msg='images not same shape')
+
+            for ix in range(cleanIm.shape[0]):
+                for iy in range(noisyIm.shape[1]):
+                    mean = cleanIm[ix, iy]
+                    var = mean/gain + readnoise/(gain*gain)
+                    totalVar += numpy.power(noisyIm[ix, iy] - mean,2)/var
+                    totalMean += mean
+                    ct += 1.0
+
+
+            totalVar = totalVar/ct
+            totalMean = totalMean/ct
+
+            if totalMean>=100.0:
+                countedImages += 1
+                self.assertTrue(numpy.abs(totalVar-1.0) < 0.05)
```

Comment:
I have set the seed on the noise generating routine.

There is a random number generator internal to GalSim that is used to generate both the noiseless and the noisy images (for each object, photons are randomly drawn from the SED and shot at the image).  I do not see hot to access this generator to seed it.